src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -219,12 +219,19 @@ public static bool IsAsyncDisposeImplementation([NotNullWhen(returnValue: true)]
                 method.IsImplementationOfInterfaceMethod(null, iAsyncDisposable, "DisposeAsync");
         }
 
+        /// <summary>
+        /// Checks the name of the method of an implicit interface implementation (so, as is) or an explicit interface implementation (which also contains the interface as a prefix). 
+        /// </summary>
+        private static bool IsImplicitOrExplicitInterfaceImplementationName(this IMethodSymbol method, string name, string interfacePrefix)
+            => (method.Name == name && method.MethodKind is MethodKind.Ordinary) ||
+            (method.Name == $"{interfacePrefix}.{name}" && method.MethodKind is MethodKind.ExplicitInterfaceImplementation);
+
         /// <summary>
         /// Checks if the given method has the signature "void Dispose()".
         /// </summary>
         private static bool HasDisposeMethodSignature(this IMethodSymbol method)
         {
-            return method.Name == "Dispose" && method.MethodKind == MethodKind.Ordinary &&
+            return method.IsImplicitOrExplicitInterfaceImplementationName("Dispose", "System.IDisposable") &&
```

Comment:
Can we use the below constant string instead of hardcoding it here: https://github.com/dotnet/roslyn-analyzers/blob/c2bc164fe991d87662dc5b6d3aa94bf276392241/src/Utilities/Compiler/WellKnownTypeNames.cs#L228